File: src/archive/22112025find_minimum_operations.go
package main

import (
	"bun-revolution/src/libgo"
)

func MinimumOperations(nums []int) int {
	// minimumOperations := 0
	// for _, num := range nums {
	//  if num%3 != 0 {
	//      minimumOperations++
	//  }
	// }
	// return minimumOperations
	filtered := libgo.Filter(nums, func(num int) bool {
		return num%3 != 0
	})
	return len(filtered)
}

File: src/archive/22112025find_minimum_operations.ts
function minimumOperations(nums: number[]): number {
    let numberOfOperations = 0

    nums.forEach(n => {
        if (n % 3 !== 0) {
            if ((n + 1) % 3 === 0) {
                numberOfOperations += 1
            } else if ((n - 1) % 3 === 0) {
                numberOfOperations += 1
            }
        }
    })
    console.log('numberOfOperations', { numberOfOperations })
    return numberOfOperations
}

minimumOperations([1, 2, 3, 4, 5])
minimumOperations([3,6,9])

// function minimumOperations(nums: number[]): number {
//    return nums.filter((num)=> num % 3 !== 0).length
// }; 